### PR TITLE
GFC-653 Value rounding

### PR DESCRIPTION
**Pull request by DarylMConway**
changed big to bignumber

changed big to bignumber for all mathematical equations

changed big to bignumber within the rounding method and got rid of unused oformat functions

created rounding mode and then sent to format

deleted un-used java rounding match statement

reverted all forms of roundingMode input to be under text not number

implemented a rounding functionality for: number, positive number and sterling

### CODE REVIEW

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/HasExpr.scala [hunk 1]
```diff
@@ -17,19 +17,25 @@
 package uk.gov.hmrc.gform.sharedmodel.formtemplate
 
 sealed trait ExprCardinality
-case class SingleExpr(expr: Expr) extends ExprCardinality
+case class SingleExpr(expr: Expr, roundingMode: Option[RoundingMode]) extends ExprCardinality
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/HasExpr.scala]: Putting `roundingMode` into this class is mixing unrelated concerns. We should move extraction of `roundingMode` from `ComponentType` directly when it is needed ie.  into `Javascript.scala`.

File: app/uk/gov/hmrc/gform/models/helpers/Javascript.scala [hunk 2]
```diff
@@ -159,11 +161,22 @@ object Javascript {
...
+      case Some(RoundingMode.Down)     => "ROUND_DOWN"
+      case Some(RoundingMode.Ceiling)  => "ROUND_CEIL"
+      case Some(RoundingMode.Floor)    => "ROUND_FLOOR"
+      case Some(RoundingMode.HalfUp)   => "ROUND_HALF_UP"
+      case Some(RoundingMode.HalfDown) => "ROUND_HALF_DOWN"
+      case Some(RoundingMode.HalfEven) => "ROUND_HALF_EVEN"
+      case _                           => "ROUND_DOWN"
+    }
     val elementId = field.id
+    val x = field
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/models/helpers/Javascript.scala]: This doesn't seems needed.

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/HasExpr.scala (hunk 1 above)
Comment (DarylMConway) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/HasExpr.scala]: roundingMode has been moved to Javascript.scala

File: app/uk/gov/hmrc/gform/models/helpers/Javascript.scala (hunk 2 above)
Comment (DarylMConway) [reply app/uk/gov/hmrc/gform/models/helpers/Javascript.scala]: line has been removed

**Pull request merged**